prime-angular-frontend/src/app/core/resources/health-authority-resource.service.ts

```diff
@@ -163,6 +166,30 @@ export class HealthAuthorityResource {
       );
   }
 
+  public createOrganizationAgreementDocument(healthAuthorityId: number, documentGuid: string): Observable<BaseDocument> {
+    const params = this.apiResourceUtilsService.makeHttpParams({ documentGuid });
+    return this.apiResource.put<BaseDocument>(`health-authorities/${healthAuthorityId}/organization-agreement`, null, params)
+      .pipe(
+        map((response: ApiHttpResponse<BaseDocument>) => response.result),
+        catchError((error: any) => {
+          this.logger.error('[Core] HealthAuthorityResource::createOrganizationAgreementDocument error has occurred: ', error);
+          throw error;
+        })
+      );
+  }
+
+  public getOrganizationAgreementDocumentToken(healthAuthorityId: number): Observable<string> {
+    return this.apiResource.get<string>(`health-authorities/${healthAuthorityId}/organization-agreement/token`)
+      .pipe(
+        map((response: ApiHttpResponse<string>) => response.result),
+        catchError((error: any) => {
+          this.toastService.openErrorToast('Organization Agreeement token could not be Retrieved');
```

Comment:
Extra "e" in "Agreeement" and incorrect casing in "Retrieved"

Reply:
I'm always worried when i write agreeeeeeements

Reply:
oops accidently clicked resolve